pkg/controller/appdefinition/serviceaccount.go

```diff
@@ -11,43 +11,45 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func addServiceAccount(appInstance *v1.AppInstance, resp router.Response) {
-	resp.Objects(toServiceAccount(appInstance)...)
-}
-
-func toRules(rules []v1.PolicyRule) (result []rbacv1.PolicyRule) {
-	for _, rule := range rules {
-		result = append(result, (rbacv1.PolicyRule)(rule))
-	}
-	return
-}
-func toServiceAccount(appInstance *v1.AppInstance) (result []kclient.Object) {
-	labelMap := labels.Merge(labels.Managed(appInstance), labels.GatherScoped("", "", appInstance.Status.AppSpec.Labels,
-		nil, appInstance.Spec.Labels))
+func addServiceAccounts(appInstance *v1.AppInstance, resp router.Response) {
+	labelMap := labels.Merge(labels.Managed(appInstance), labels.GatherScoped("", "", appInstance.Status.AppSpec.Labels, nil, appInstance.Spec.Labels))
```

Comment:
Labels and annotations are "gathered" from the appImage and the spec and applied to relevant resources. The labels can have different "scopes", see: https://docs.acorn.io/running/labels

When there was one service account for an entire app, it made sense for that service account to get just the "globally" labels and annotations. Now that they're specific to the acorn container, it makes sense to have them match just what applies to the deployment. Fortunately, there's helper functions. You can see them in use here:
https://github.com/acorn-io/acorn/blob/main/pkg/controller/appdefinition/deploy.go#L570 (labels)
https://github.com/acorn-io/acorn/blob/main/pkg/controller/appdefinition/deploy.go#L574 (annotations)

Reply:
Okay, now we are using the same labels and annotations as the deployment in the most recent push.